alg/md5.c

```diff
@@ -294,11 +294,11 @@ MD5_Buf(const void * in, size_t len, uint8_t digest[16])
  * Initialize the HMAC-MD5 context ${ctx} with ${Klen} bytes of key from ${K}.
  */
 void
-HMAC_MD5_Init(HMAC_MD5_CTX * ctx, const void * _K, size_t Klen)
+HMAC_MD5_Init(HMAC_MD5_CTX * ctx, const void * arg_K, size_t Klen)
```

Comment:
I've gone back and forth about what to rename this to.  `orig_K`? `var_K`?  `K_`? `p_K`?

In the end, I couldn't think of any reason to choose one over the others, so I figured I'd pick one and open the PR.

Reply:
A similar question, although not part of this particular PR: what naming scheme should we use for functions which are reserved identifiers?  For example, `_SHA256_Update: should that be `p_SHA256_Update`?  `SHA256_Update_`?  `real_SHA256_Update`?  `SHA256_Update_with_tmp`?

Reply:
Another possibility to replace `_SHA256_Update`: we could use `_sha256_Update`.  Of course we want the public API to be in mostly upper-case letters to match the RFC, but the underscore-prepended version is strictly an internal function, so it would be fine to use lower-case for it.